src/braket/pulse/waveforms.py

```diff
@@ -501,6 +502,166 @@ def _from_calibration_schema(waveform_json: dict) -> GaussianWaveform:
         return GaussianWaveform(**waveform_parameters)
 
 
+class ErfSquareWaveform(Waveform, Parameterizable):
+    """A square waveform with smoothed edges."""
+
+    def __init__(
+        self,
+        length: Union[float, FreeParameterExpression],
+        width: Union[float, FreeParameterExpression],
+        sigma: Union[float, FreeParameterExpression],
+        amplitude: Union[float, FreeParameterExpression] = 1,
+        zero_at_edges: bool = False,
+        id: Optional[str] = None,
+    ):
+        r"""Initializes a `ErfSquareWaveform`.
+
+        .. math:: (\text{step}((t-t_1)/sigma) + \text{step}(-(t-t_2)/sigma) - 1)
+
+        where :math:`\text{step}(t)` is the rounded step function defined as
+        :math:`(erf(t)+1)/2` and :math:`t_1` and :math:`t_2` are the timestamps at the half
+        height. The waveform is scaled such that its maximum is equal to `amplitude`.
+
+        Args:
+            length (Union[float, FreeParameterExpression]): Duration (in seconds) from the start
+                to the end of the waveform.
+            width (Union[float, FreeParameterExpression]): Duration (in seconds) between the
+                half height of the two edges.
+            sigma (Union[float, FreeParameterExpression]): A characteristic time of how quickly
+                the edges rise and fall.
+            amplitude (Union[float, FreeParameterExpression]): The amplitude of the waveform
+                envelope. Defaults to 1.
+            zero_at_edges (bool): Whether the waveform is scaled such that it has zero value at the
+                edges. Defaults to False.
+            id (Optional[str]): The identifier used for declaring this waveform. A random string of
+                ascii characters is assigned by default.
+        """
+        self.length = length
+        self.width = width
+        self.sigma = sigma
+        self.amplitude = amplitude
+        self.zero_at_edges = zero_at_edges
+        self.id = id or _make_identifier_name()
+
+    def __repr__(self) -> str:
+        return (
+            f"ErfSquareWaveform('id': {self.id}, 'length': {self.length}, "
+            f"'width': {self.width}, 'sigma': {self.sigma}, 'amplitude': {self.amplitude}, "
+            f"'zero_at_edges': {self.zero_at_edges})"
```

Comment:
`__repr__()` is supposed to return a valid Python string which can be used to construct the object:
```suggestion
            f"ErfSquareWaveform(id='{self.id}', length={self.length}, "
            f"width={self.width}, sigma={self.sigma}, amplitude={self.amplitude}, "
            f"zero_at_edges={self.zero_at_edges}"
```
but I see that existing classes in this file also have this problem.

Reply:
The repr of all existing waveforms all use the the same convention. I will keep erf square the same.